### PR TITLE
Expose AI chat flag on Android

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/32807

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

[device-2023-09-05-170208.webm](https://github.com/brave/brave-core/assets/12011303/e7409e94-7ba5-4a04-b77f-5943dae0b380)

### CODE REVIEW

File: browser/ui/webui/ai_chat/ai_chat_ui.cc [hunk 1]
```diff
@@ -8,22 +8,25 @@
...
-#include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/webui/webui_util.h"
 #include "components/grit/brave_components_resources.h"
 #include "components/prefs/pref_service.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_ui_data_source.h"
 #include "content/public/common/url_constants.h"
 
+#if !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h"
```
Comment (SergeyZhukovsky) [inline browser/ui/webui/ai_chat/ai_chat_ui.cc:26]: I'm going to work on a handler in the follow up issue as we don't have that on Android `browser_->tab_strip_model()`

File: components/resources/brave_components_resources.grd [hunk 2]
```diff
@@ -99,8 +99,8 @@
       <if expr="not is_android">
         <part file="brave_shields_resources.grdp" />
         <part file="brave_private_new_tab_resources.grdp" />
-        <part file="../ai_chat/resources/page/ai_chat_ui_resources.grdp" />
       </if>
+      <part file="../ai_chat/resources/page/ai_chat_ui_resources.grdp" />
```
Comment (SergeyZhukovsky) [inline components/resources/brave_components_resources.grd:103]: I'm going to check what particular string do we need on Android when handlers and the whole WebUI is visible.

File: components/resources/brave_components_strings.grd [hunk 3]
```diff
@@ -829,11 +829,11 @@
       <part file="brave_news_strings.grdp" />
       <part file="commands.grdp" />
       <part file="speedreader_strings.grdp" />
+      <part file="ai_chat_prompts.grdp" />
+      <part file="ai_chat_ui_strings.grdp" />
```
Comment (SergeyZhukovsky) [inline components/resources/brave_components_strings.grd:833]: the same about strings as in the comment above.

File: browser/ui/webui/ai_chat/ai_chat_ui.cc (hunk 1 above)
Comment (SergeyZhukovsky) [reply browser/ui/webui/ai_chat/ai_chat_ui.cc:26]: the follow up issue is https://github.com/brave/brave-browser/issues/32808

**Review by mkarolin — APPROVED** [review]
`strings` and `chromium_src` LGTM

**Review by AlexeyBarabash — APPROVED** [review]
++

File: components/resources/brave_components_resources.grd (hunk 2 above)
Comment (petemill) [reply components/resources/brave_components_resources.grd:103]: all of them I'd guess

**Review by thypon — COMMENTED** [review]
security gtg

**Pull request merged**